urlcaef331f4e_dreamable_cra_tgz-pJ8-coho_lp_project_LPProjectJ8/src/coho/common/number/ScaleType.java

```diff
@@ -0,0 +1,309 @@
+		}
+	}
+	
+	
+	public static ScaleNumber promoteOp(ScaleNumber n1, ScaleNumber n2, CohoNumber.ArithOp op){
+		ScaleNumber[] operands = promote(n1,n2);
+		ScaleType t = operands[0].type();
+		if(t == CohoBoolean.type){
+			CohoBoolean op1 = (CohoBoolean)operands[0];
+			CohoBoolean op2 = (CohoBoolean)operands[1];
+			switch(op){
+			case ADD:
+				return op1.add(op2);
+			case SUB:
+				return op1.sub(op2);
+			case MULT:
+				return op1.mult(op2);
+			case DIV:
+				return op1.div(op2);
+			case MAX:
+				return op1.max(op2);
+			case MIN:
+				return op1.min(op2);
+			case CMP:
+				return CohoInteger.create(op1.compareTo(op2));
+			default:
+				throw new UnsupportedOperationException(op.name());
+			}
+		}else if(t == CohoInteger.type){
+			CohoInteger op1 = (CohoInteger)operands[0];
+			CohoInteger op2 = (CohoInteger)operands[1];
+			switch(op){
+			case ADD:
+				return op1.add(op2);
+			case SUB:
+				return op1.sub(op2);
+			case MULT:
+				return op1.mult(op2);
+			case DIV:
+				return op1.div(op2);
+			case MAX:
+				return op1.max(op2);
+			case MIN:
+				return op1.min(op2);
+			case CMP:
+				return CohoInteger.create(op1.compareTo(op2));
+			default:
+				throw new UnsupportedOperationException(op.name());
+			}			
+		}else if(t == CohoLong.type){
+			CohoLong op1 = (CohoLong)operands[0];
+			CohoLong op2 = (CohoLong)operands[1];
+			switch(op){
+			case ADD:
+				return op1.add(op2);
+			case SUB:
+				return op1.sub(op2);
+			case MULT:
+				return op1.mult(op2);
+			case DIV:
+				return op1.div(op2);
+			case MAX:
+				return op1.max(op2);
+			case MIN:
+				return op1.min(op2);
+			case CMP:
+				return CohoInteger.create(op1.compareTo(op2));
+			default:
+				throw new UnsupportedOperationException(op.name());
+			}			
+		}else if(t == CohoDouble.type){
+			CohoDouble op1 = (CohoDouble)operands[0];
+			CohoDouble op2 = (CohoDouble)operands[1];
+			switch(op){
+			case ADD:
+				return op1.add(op2);
+			case SUB:
+				return op1.sub(op2);
+			case MULT:
+				return op1.mult(op2);
+			case DIV:
+				return op1.div(op2);
+			case MAX:
+				return op1.max(op2);
+			case MIN:
+				return op1.min(op2);
+			case CMP:
+				return CohoInteger.create(op1.compareTo(op2));
+			default:
+				throw new UnsupportedOperationException(op.name());
+			}			
+		}else if(t == CohoAPR.type){
+			CohoAPR op1 = (CohoAPR)operands[0];
+			CohoAPR op2 = (CohoAPR)operands[1];
+			switch(op){
+			case ADD:
+				return op1.add(op2);
+			case SUB:
+				return op1.sub(op2);
+			case MULT:
+				return op1.mult(op2);
+			case DIV:
+				return op1.div(op2);
+			case MAX:
+				return op1.max(op2);
+			case MIN:
+				return op1.min(op2);
+			case CMP:
+				return CohoInteger.create(op1.compareTo(op2));
+			default:
+				throw new UnsupportedOperationException(op.name());
+			}			
+		}else
+			throw new RuntimeException("Unknow ScaleType"+t);
+    	
+	}
+	public static ScaleNumber promoteOp(ScaleNumber n1, Number n2, CohoNumber.ArithOp op){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 104 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.